sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -770,6 +770,12 @@ object SQLConf {
     .intConf
     .createWithDefault(200)
 
+  val THRIFTSERVER_RESULT_ESCAPE_STRUCT_STRING =
+    buildConf("spark.sql.thriftserver.result.escapeStructString.enabled")
+    .doc("When true, escape string when needed to ensure the result returned is a valid json.")
+    .booleanConf
+    .createWithDefault(false)
```

Comment:
Is this a bug or a hive compatibility issue? If so, we might be able to set true by default? Then, update the migration guide.

Reply:
I think this should be a bug.  I can't think of any other reasons to return a non-json string here.

Reply:
cc: @wangyum @dongjoon-hyun